handlers/upload_service.go

```diff
@@ -111,8 +114,24 @@ func (us *UploadServiceImpl) ProcessUpload(r *UploadRequest) error {
 }
 
 // Retrieves an error if the signature is invalid, of if the signature does not corresponds to the given key and message
-func validateSignature(a data.Authorization, m Metadata) error {
+func (us *UploadServiceImpl) validateSignature(a data.Authorization, m Metadata) error {
 	log.Debugf("Validating signature: %s", m.Signature)
+
+	// ERC 1654 support https://github.com/ethereum/EIPs/issues/1654
+	// We need to validate against a contract address whether this is ok or not?
+	if len(m.Signature) > 66 {
+		signature := m.Signature
+		address := m.PubKey
+		msg := fmt.Sprintf("%s.%d", m.Value, m.Timestamp)
+		valid, err := us.rpc.ValidateDapperSignature(address, msg, signature)
+		if err != nil {
```

Comment:
Log the error, just to keep track of the Dapper case

Reply:
Added logs at the service level